programs/uxd/tests/integration_tests/suites/test_mint_and_redeem.rs

```diff
@@ -77,16 +77,15 @@ async fn test_mint_and_redeem() -> Result<(), program_test_context::ProgramTestE
     // Useful amounts used during testing scenario
     let amount_we_use_as_supply_cap = ui_amount_to_native_amount(1000, redeemable_mint_decimals);
 
-    let amount_of_collateral_airdropped_to_user =
-        ui_amount_to_native_amount(1000, collateral_mint_decimals);
-
     let amount_for_first_mint = ui_amount_to_native_amount(100, collateral_mint_decimals);
     let amount_for_second_mint = ui_amount_to_native_amount(200, collateral_mint_decimals);
 
     let amount_for_first_redeem = ui_amount_to_native_amount(20, redeemable_mint_decimals);
     let amount_for_second_redeem = ui_amount_to_native_amount(120, redeemable_mint_decimals);
     let amount_for_third_redeem = ui_amount_to_native_amount(10, redeemable_mint_decimals);
 
+    let amount_of_collateral_airdropped_to_user = amount_for_first_mint + amount_for_second_mint; // Just enough money to mint
```

Comment:
added a test, so that we dont have any collateral left before we start redeeming (detect the above case)